x/marker/keeper/keeper_test.go

```diff
@@ -1415,3 +1415,114 @@ func TestRemoveIsSendEnabledEntries(t *testing.T) {
 	assert.True(t, app.BankKeeper.IsSendEnabledDenom(ctx, "testcoin"), "should not exist in table therefore default to true")
 
 }
+
+func TestMsgUpdateSendDenyListRequest(t *testing.T) {
+	app := simapp.Setup(t)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+	server := markerkeeper.NewMsgServerImpl(app.MarkerKeeper)
+
+	authority := authtypes.NewModuleAddress(govtypes.ModuleName)
+
+	authUser := testUserAddress("test")
+	notAuthUser := testUserAddress("test1")
+
+	notRestrictedMarker := types.NewEmptyMarkerAccount(
+		"not-restricted-marker",
+		authUser.String(),
+		[]types.AccessGrant{})
+
+	err := app.MarkerKeeper.AddMarkerAccount(ctx, notRestrictedMarker)
+	require.NoError(t, err)
+
+	rMarkerDenom := "restricted-marker"
+	rMarkerAcct := authtypes.NewBaseAccount(types.MustGetMarkerAddress(rMarkerDenom), nil, 0, 0)
+	app.MarkerKeeper.SetMarker(ctx, types.NewMarkerAccount(rMarkerAcct, sdk.NewInt64Coin(rMarkerDenom, 1000), authUser, []types.AccessGrant{{Address: authUser.String(), Permissions: []types.Access{types.Access_Transfer}}}, types.StatusFinalized, types.MarkerType_RestrictedCoin, true, false, false, []string{}))
+
+	rMarkerGovDenom := "restricted-marker-gov"
+	rMarkerGovAcct := authtypes.NewBaseAccount(types.MustGetMarkerAddress(rMarkerGovDenom), nil, 0, 0)
+	app.MarkerKeeper.SetMarker(ctx, types.NewMarkerAccount(rMarkerGovAcct, sdk.NewInt64Coin(rMarkerGovDenom, 1000), authUser, []types.AccessGrant{{Address: authUser.String(), Permissions: []types.Access{}}}, types.StatusFinalized, types.MarkerType_RestrictedCoin, true, true, false, []string{}))
+
+	denyAddrToRemove := testUserAddress("denyAddrToRemove")
+	app.MarkerKeeper.AddSendDeny(ctx, rMarkerAcct.GetAddress(), denyAddrToRemove)
+	require.True(t, app.MarkerKeeper.IsSendDeny(ctx, rMarkerAcct.GetAddress(), denyAddrToRemove), rMarkerDenom+" should have added address to deny list "+denyAddrToRemove.String())
+
+	denyAddrToAdd := testUserAddress("denyAddrToAdd")
+
+	denyAddrToAddGov := testUserAddress("denyAddrToAddGov")
+
+	testCases := []struct {
+		name             string
+		updateMsgRequest types.MsgUpdateSendDenyListRequest
+		expectedReqAttr  []string
+		expectedError    string
```

Comment:
Here too with the shorter names thing. Also, `expectedReqAttr` isn't used and should be removed.

But you could call the fields just `name`, `msg`, and `expErr`, without losing any readability.
You could also use a `*types.MsgUpdateSendDenyListRequest` since that's what the `UpdateSendDenyList` function accepts, and you're always using the constructor, which returns the reference which you then have to dereference for the test case and re-reference for the call in the test.

Also, I'm really not a fan of using the constructors in unit tests, specially when they're so simple. My problem with them is that, when we add a field to the message, we then have to add it to the constructor, and then we have to update all these test cases to include a value for the new field. In 99% of the cases, we just want the zero value, or one specific value, and it'd be better if we didn't even have to make a change to each test case in order for the tests to continue to pass. Plus, creating your own private constructor in this test (e.g. `newMsg := func(...)`) let's you use a smaller name, making it easier to read the test case definitions; it helps with the signal (msg values) to noise (the name of the constructor) ratio.